src/library/scala/collection/immutable/ChampHashMap.scala

```diff
@@ -170,16 +197,16 @@ private final class BitmapIndexedMapNode[K, +V](val dataMap: Int, val nodeMap: I
 
     if ((dataMap & bitpos) != 0) {
       val index = indexFrom(dataMap, mask, bitpos)
-      val payload = this.getPayload(index)
-      return if (key == payload._1) Option(payload._2) else Option.empty
```

Comment:
@msteindorfer Not sure if you're aware, but `Option(null)` == `None`. That's not what's wanted here, as we can see from:

```
Welcome to Scala 2.13.0-M4 (Java HotSpot(TM) 64-Bit Server VM, Java 1.8.0_172).
Type in expressions for evaluation. Or try :help.

scala> val map = collection.immutable.ChampHashMap[Any, Any]()
map: scala.collection.immutable.ChampHashMap[Any,Any] = ChampHashMap()

scala> map.updated("", null).get("")
res0: Option[Any] = None

scala> val map = collection.immutable.HashMap[Any, Any]()
map: scala.collection.immutable.HashMap[Any,Any] = HashMap()

scala> map.updated("", null).get("")
res1: Option[Any] = Some(null)
```

Reply:
My mistake. I used `Option.empty` wrongly, thought it construct `None`.

Reply:
It's `Option.apply` which is something of a trap. It has the same semantics Java's `Optional.ofNullable`.

```
scala> Option.empty == None
res1: Boolean = true

scala> Option.apply("") == Some("")
res2: Boolean = true

scala> Option.apply(null) == null
res3: Boolean = false
```